### PR TITLE
tailcfg,ipn/ipnlocal: add hostinfo field to replace service entry

### PR DESCRIPTION
**Pull request by raggi**
@bradfitz I assume this would be the preferred approach?

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -650,9 +651,9 @@ type Service struct {
 	//        being a DNS proxy (when the node is an exit
 	//        node). For this service, the Port number is really
 	//        the version number of the service.
-	//     * "app-connector": the local app-connector service is
-	//        available. For this service, the Port number is
-	//        really the version number of the service.
+	//     * "app-connector": (deprecated) the local app-connector
+	//        service is available. For this service, the Port number
+	//        is really the version number of the service.
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:656]: btw, we used to have N version numbers in the client and consolidated to using CapVer for everything. (it used to be named MapRequest.Version or "mapver" but we renamed it mostly everywhere to "capver" to be more general)

File: tailcfg/tailcfg.go [hunk 2]
```diff
@@ -624,11 +624,12 @@ func (h *Hostinfo) CheckRequestTags() error {
...
-	UDP          = ServiceProto("udp")
-	PeerAPI4     = ServiceProto("peerapi4")
-	PeerAPI6     = ServiceProto("peerapi6")
-	PeerAPIDNS   = ServiceProto("peerapi-dns-proxy")
+	TCP        = ServiceProto("tcp")
+	UDP        = ServiceProto("udp")
+	PeerAPI4   = ServiceProto("peerapi4")
+	PeerAPI6   = ServiceProto("peerapi6")
+	PeerAPIDNS = ServiceProto("peerapi-dns-proxy")
+	// Deprecated: use the field on HostInfo instead.
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:632]: toss a date/capver/issue URL on this? was this ever in a released client? if so, something like "used until 1.52.x".

File: ipn/ipnlocal/local.go [hunk 3]
```diff
@@ -3270,6 +3271,11 @@ func (b *LocalBackend) blockEngineUpdates(block bool) {
 // b.mu must be held.
 func (b *LocalBackend) reconfigAppConnectorLocked(nm *netmap.NetworkMap, prefs ipn.PrefsView) {
 	const appConnectorCapName = "tailscale.com/app-connectors"
+	defer func() {
+		if b.hostinfo != nil {
+			b.hostinfo.AppConnector.Set(b.appConnector != nil)
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:3276]: the hostinfo handling in LocalBackend changed a fair bit recently to be slightly less dumpster fire-y than it historically has been but it's still pretty treacherous.

Note that Start blows this all away. Not sure whether you're restoring it all from that path or not.

**Review by bradfitz — APPROVED** [review]

File: tailcfg/tailcfg.go (hunk 1 above)
Comment (raggi) [reply tailcfg/tailcfg.go:656]: Yup, that makes sense. That was added mostly for consistency m.

File: tailcfg/tailcfg.go (hunk 2 above)
Comment (raggi) [reply tailcfg/tailcfg.go:632]: It hasn't been released yet, and we can avoid retaining it. Once we've rolled this through control I'll come back and remove it. Currently connectors are only supported in unstable and we can put a capver gate on them when we change that

File: ipn/ipnlocal/local.go (hunk 3 above)
Comment (raggi) [reply ipn/ipnlocal/local.go:3276]: yeah, there's a set in start following the same pattern too

**Pull request merged**